app/scripts/modules/amazon/src/domain/IAmazonLoadBalancerSourceData.ts

```diff
@@ -0,0 +1,127 @@
+import { ILoadBalancerSourceData } from '@spinnaker/core';
+
+export interface IAmazonLoadBalancerServerGroupSourceData {
+  detachedInstances: string[];
+  instances: IAmazonLoadBalancerInstanceSourceData[];
+  isDisabled: boolean;
+  name: string;
+  region: string;
+}
+
+export interface IAmazonInstanceHealthSourceData {
+  type: string;
+  state: 'InService' | 'OutOfService' | 'Unknown' | 'healthy';
+  reasonCode: 'ELB' | 'Instance' | 'N/A';
+  description: string;
+}
+
+export interface IClassicListenerSourceData {
+  instancePort: number;
+  instanceProtocol: string;
+  loadBalancerPort: number;
+  protocol: string;
+};
+
+export interface IAmazonLoadBalancerSourceData extends ILoadBalancerSourceData {
+  account: string;
+  availabilityZones: string[];
+  cloudProvider: string;
+  createdTime: number;
+  dnsname: string;
+  loadBalancerName: string;
+  name: string;
+  region: string;
+  scheme: 'internal' | 'internet-facing';
+  securityGroups: string[];
+  serverGroups: IAmazonLoadBalancerServerGroupSourceData[];
+  subnets: string[];
+  type: string;
+  vpcId: string;
+  vpcid?: string;
+};
+
+export interface IAmazonLoadBalancerInstanceSourceData {
+  id: string;
+  zone: string;
+  health: IAmazonInstanceHealthSourceData;
+}
+
+export interface IAmazonTargetGroupSourceData {
+  account: string;
+  attributes: {
+    'deregistration_delay.timeout_seconds': number;
+    'stickiness.enabled': boolean;
+    'stickiness.lb_cookie.duration_seconds': number;
+    'stickiness.type': 'lb_cookie';
+  };
+  cloudProvider: string;
+  healthCheckIntervalSeconds: number;
+  healthCheckPath: string;
+  healthCheckPort: string;
+  healthCheckProtocol: string;
+  healthCheckTimeoutSeconds: number;
+  healthyThresholdCount: number;
+  instances: string[];
+  loadBalancerNames: string[];
+  matcher: {
+    httpCode: string;
+  };
+  name: string;
+  port: number;
+  protocol: string;
+  region: string;
+  serverGroups: IAmazonLoadBalancerServerGroupSourceData[];
+  targetGroupArn: string;
+  targetGroupName: string;
+  type: string;
+  unhealthyThresholdCount: number;
+  vpcId: string;
+}
+
+export interface IApplicationLoadBalancerSourceData extends IAmazonLoadBalancerSourceData {
+  ipAddressType: 'ipv4' | 'dualstack';
+  listeners: {
+    defaultActions: {
+      targetGroupName: string;
+      type: 'forward';
+    }[];
```

Comment:
I like the nested declaration approach (vs. declaring a bunch of extra interfaces that are only used in one place and never referenced).